#Estrutura de Repetição

**Faça um programa que peça uma nota, entre zero e dez. Mostre uma mensagem caso o valor seja inválido e continue pedindo até que o usuário informe um valor válido.**

In [ ]:
nota = int(input('Informe uma nota, de 0 a 10: '))

while not (nota >= 0 and nota <= 10):
  print('Valor inválido')
  nota = int(input('Informe uma nota válida, de 0 a 10: '))
print (f'Nota {nota} válida')


Informe uma nota, de 0 a 10: 12
Valor inválido
Informe uma nota válida, de 0 a 10: 3
Nota 3 válida


**Faça um programa que leia um nome de usuário e a sua senha e não aceite a senha igual ao nome do usuário, mostrando uma mensagem de erro e voltando a pedir as informações.**

In [ ]:
x = True
nome = input('Digite seu nome: ')

while x:
  senha = input('Digite uma senha alfabética: ')
  if nome == senha:
    print('A senha é igual ao seu nome!') 
  else:
    x = False
print('Senha cadastrada!')

Digite seu nome: mari
Digite uma senha alfabética: mari
A senha é igual ao seu nome!
Digite uma senha alfabética: ana
Senha cadastrada!


**Faça um programa que leia e valide as seguintes informações:**
1. Nome: maior que 3 caracteres;
2. Idade: entre 0 e 150;
3. Salário: maior que zero;
4. Sexo: 'f' ou 'm';
5. Estado Civil: 's', 'c', 'v', 'd';


In [ ]:
nome = input('Informe seu nome: ')

while len(nome) <= 3:
  print ('Nome invalido')
  nome = input('Informe seu nome: ')

idade = int(input('Informe a sua idade: '))

while not (idade >= 0 and idade <= 100):
  print('Idade inválida')
  idade = int(input('Informe a sua idade: '))

salario = float(input('Informe o salário: '))

while not (salario > 0):
  print('Salário inválido')
  salario = float(input('Informe o salário: '))

sexo = input('Informe o sexo (f ou m): ')

while not (sexo == 'f' or sexo == 'm'):
  print('Sexo inválido')
  sexo = input('Informe o sexo: ')

estado_civil = input('Informe o estado civil (s, c, v ou d): ')
 
while not (estado_civil == 's' or estado_civil == 'c' or estado_civil == 'v' or estado_civil =='d'):
  print('Estado civil inválido')
  estado_civil = input('Informe o estado civil (s, c, v ou d): ')

print('Dados válidos')

Informe seu nome: ana
Nome invalido
Informe seu nome: maria
Informe a sua idade: 120
Idade inválida
Informe a sua idade: 40
Informe o salário: 0
Salário inválido
Informe o salário: 2000
Informe o sexo (f ou m): g
Sexo inválido
Informe o sexo: f
Informe o estado civil (s, c, v ou d): k
Estado civil inválido
Informe o estado civil (s, c, v ou d): c
Dados válidos


**Supondo que a população de um país A seja da ordem de 80000 habitantes com uma taxa anual de crescimento de 3% e que a população de B seja 200000 habitantes com uma taxa de crescimento de 1.5%. Faça um programa que calcule e escreva o número de anos necessários para que a população do país A ultrapasse ou iguale a população do país B, mantidas as taxas de crescimento.**

In [ ]:
count = 0
a = 80000
b = 200000

while a <= b:
  count += 1
  a = a + (a * 0.03)
  b = b + (b * 0.015)

print(f'A população do país A ultrapassa a população do país B em {count} anos')

A população do país A ultrapassa a população do país B em 63 anos


**Altere o programa anterior permitindo ao usuário informar as populações e as taxas de crescimento iniciais. Valide a entrada e permita repetir a operação.**


In [ ]:
count = 0
a = int(input('Informe a população do país A: '))
crescimento_a = (float(input('Informe a taxa de crescimento do país A: ')))
b = int(input('Informe a população do país B: '))
crescimento_b = (float(input('Informe a taxa de crescimento do país B: ')))

while a <= b:
  count += 1
  a = a + (a * crescimento_a)
  b = b + (b * crescimento_b)

print(f'A população do país A ultrapassa a população do país B em {count} anos')

Informe a população do país A: 80000
Informe a taxa de crescimento do país A: 200000
Informe a população do país B: 90000
Informe a taxa de crescimento do país B: 150000
A população do país A ultrapassa a população do país B em 1 anos


**Faça um programa que imprima na tela os números de 1 a 20, um abaixo do outro. Depois modifique o programa para que ele mostre os números um ao lado do outro.**

In [ ]:
#um abaixo do outro
for i in range(1, 21):
  print(i)

#um ao lado do outro
for i in range(1,21):
  print(i, end='')

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
1234567891011121314151617181920

**Faça um programa que leia 5 números e informe o maior número.**

In [ ]:
maior = -999999
count = 1

while count <= 5:
  numero = int(input('Informe um número: '))
  if numero > maior:
    maior = numero
  count += 1

print(maior)

Informe um número: 3
Informe um número: 576
Informe um número: 34
Informe um número: 976
Informe um número: 124
976


**Faça um programa que leia 5 números e informe a soma e a média dos números.**

In [ ]:
count = 1
soma = 0

while count <= 5:
  numero = int(input('Informe um número: '))
  soma += numero
  count += 1

media = soma / 5

print(f'A soma dos números é {soma}')
print(f'A média dos números é {media}')

Informe um número: 23
Informe um número: 56
Informe um número: 23
Informe um número: 86
Informe um número: 26
A soma dos números é 214
A média dos números é 42.8


**Faça um programa que imprima na tela apenas os números ímpares entre 1 e 50.**

In [ ]:
x = 1
while x <= 50:
  if not ((x % 2) == 0):
    print(x)
  x += 1

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49


**Faça um programa que receba dois números inteiros e gere os números inteiros que estão no intervalo compreendido por eles.**

In [ ]:
num1 = int(input('Informe um número: '))
num2 = int(input('Informe outro número: '))

if num1 < num2:
  for i in range (num1+1, num2, 1):
    print(i)
else:
  for i in range (num2+1, num1, 1):
    print(i)

Informe um número: 2
Informe outro número: 8
3
4
5
6
7


**Altere o programa anterior para mostrar no final a soma dos números.**

In [ ]:
n1 = int(input('Informe um número: '))
n2 = int(input('Informe outro número: '))

soma = n1 + n2

if n1 < n2:
  for i in range (n1+1, n2, 1):
    print(i)
else:
  for i in range (n2+1, n1, 1):
    print(i)

print(f'A soma dos números {n1} e {n2} é: {soma}')

Informe um número: 2
Informe outro número: 9
3
4
5
6
7
8
A soma dos números 2 e 9 é: 11


**Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer número inteiro entre 1 a 10. O usuário deve informar de qual numero ele deseja ver a tabuada. A saída deve ser conforme o exemplo abaixo:**

```
Tabuada de 5:
5 X 1 = 5
5 X 2 = 10
...
5 X 10 = 50
```



In [ ]:
n1 = int (input ('Digite um número: '))

for n2 in range(1, 11):
  print(f'{n1} x {n2} = {(n1 * n2)}')

Digite um número: 5
5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50


**Faça um programa que peça dois números, base e expoente, calcule e mostre o primeiro número elevado ao segundo número. Não utilize a função de potência da linguagem.**

In [ ]:
base = int(input('Informe a base: '))
expoente = int(input('Informe o expoente: '))

resultado = 1
x = 1

while x <= expoente:
  resultado *= base
  x += 1

print(f'{base}  elevado a  {expoente}  é igual a {resultado}')

Informe a base: 3
Informe o expoente: 3
3  elevado a  3  é igual a 27


**Faça um programa que peça 10 números inteiros, calcule e mostre a quantidade de números pares e a quantidade de números impares.**

In [ ]:
count = 1
par = 0
impar = 0

while count <= 10:
  numero = int(input('Informe um número: '))
  if (numero % 2) == 0:
    par += 1
  else: 
    impar += 1
  count += 1

print(f'{par} números pares e {impar} números ímpares')

Informe um número: 4
Informe um número: 7
Informe um número: 345
Informe um número: 34
Informe um número: 8
Informe um número: 7
Informe um número: 9
Informe um número: 55
Informe um número: 87
Informe um número: 12
4 números pares e 6 números ímpares


**A série de Fibonacci é formada pela seqüência 1,1,2,3,5,8,13,21,34,55,... Faça um programa capaz de gerar a série até o n−ésimo termo.**

In [ ]:
termo = int(input('Informe um termo: '))
num1 = 0
num2 = 0

while (num2 <= termo):
  print(num2)
  num2 += num1
  num1 = num2 - num1

  if (num2 == 0):
    num2 += 1

Informe um termo: 55
0
1
1
2
3
5
8
13
21
34
55


**A série de Fibonacci é formada pela seqüência 0,1,1,2,3,5,8,13,21,34,55,... Faça um programa que gere a série até que o valor seja maior que 500.**

In [ ]:
num1 = 0
num2 = 0

while (num2 <= 500):
  termo =+ 1
  print(num2)
  num2 += num1
  num1 = num2 - num1
  
  if (num2 == 0):
    num2 += 1

0
1
1
2
3
5
8
13
21
34
55
89
144
233
377


**Faça um programa que calcule o fatorial de um número inteiro fornecido pelo usuário. Ex.: 5!=5.4.3.2.1=120**

In [ ]:
num = int(input('Fatorial de: '))

fatorial = 1
count = 1

while count <= num:
  fatorial *= count
  count += 1

print(fatorial)

Fatorial de: 5
120


**Faça um programa que, dado um conjunto de N números, determine o menor valor, o maior valor e a soma dos valores.**

In [ ]:
n = int(input('Informe a quantidade de números: '))

numero = int(input('Informe um número: '))
maior = numero
menor = numero

for i in range (1, n):
  numero = int(input('Informe um número: '))
  if numero > maior:
    maior = numero
  elif numero < menor:
    menor = numero

soma = maior + menor

print(f'O menor número é: {menor}')
print(f'O maior número é: {maior}')
print(f'A soma dos números {maior} e {menor} é: {soma}')

Informe a quantidade de números: 5
Informe um número: 9
Informe um número: 7
Informe um número: 12
Informe um número: 8
Informe um número: 45
O menor número é: 7
O maior número é: 45
A soma dos números 45 e 7 é: 52


**Altere o programa anterior para que ele aceite apenas números entre 0 e 1000.**

In [ ]:
n = int(input('Informe a quantidade de números: '))

numero = int(input('Informe um número entre 0 e 1000: '))
while not (numero >= 0 and numero <= 1000):
  print ('Número inválido')
  numero = int(input('Informe um número entre 0 e 1000: '))

maior = numero
menor = numero
soma = numero 

for i in range (1, n):
  numero = int(input('Informe um número entre 0 e 1000: '))

  while not (numero >= 0 and numero <= 1000):
    print ('Número inválido')
    numero = int(input('Informe um número entre 0 e 1000: '))

  if numero > maior:
    maior = numero
  elif numero < menor:
    menor = numero

  soma += numero

print(f'O menor número é: {menor}')
print(f'O maior número é: {maior}')
print(f'A soma dos números é: {soma}')

Informe a quantidade de números: 5
Informe um número entre 0 e 1000: 1000
Informe um número entre 0 e 1000: 235
Informe um número entre 0 e 1000: 245
Informe um número entre 0 e 1000: 54
Informe um número entre 0 e 1000: 3
O menor número é: 3
O maior número é: 1000
A soma dos números é: 1537


**Altere o programa de cálculo do fatorial, permitindo ao usuário calcular o fatorial várias vezes e limitando o fatorial a números inteiros positivos e menores que 16.**

In [ ]:
n = int(input('Informe a quantidade de vezes que deseja calcular: '))
count1 = 1

while count1 <= n:
  count1 += 1
  num = int(input('Fatorial de: '))

  while not (num >= 0 and num <= 16):
    print('Número inválido')
    num = int(input('Informe um número entre 0 e 16: '))

  resultado = 1
  count2 = 1

  while count2 <= num:
    resultado *= count2
    count2 += 1

  print(resultado)

Informe a quantidade de vezes que deseja calcular: 2
Fatorial de: 5
120
Fatorial de: 20
Número inválido
Informe um número entre 0 e 16: 16
20922789888000


**Faça um programa que peça um número inteiro e determine se ele é ou não um número primo. Um número primo é aquele que é divisível somente por ele mesmo e por 1.**

In [ ]:
num = int(input('Informe um número: '))
div = 0
count = 0

if (num == 0) or (num == 1):
  print('Não é primo')
else:
  for count in range(2, num):
    if (num % count) == 0:
      div += 1

  if div == 0:
    print('É primo')
  else:
    print('Não é primo')

Informe um número: 2
É primo


**Altere o programa de cálculo dos números primos, informando, caso o número não seja primo, por quais número ele é divisível.**

In [ ]:
num = int(input('Informe um número: '))
div = 0
count = 0

if (num == 0) or (num == 1):
  print('Não é primo')
else:
  for count in range(2, num):
    if (num % count) == 0:
      print(f'Divisível por: {count}')
      div += 1
      
  if div == 0:
    print('É primo')
  else:
    print('Não é primo')

Informe um número: 12
Divisível por: 2
Divisível por: 3
Divisível por: 4
Divisível por: 6
Não é primo


**Faça um programa que mostre todos os primos entre 1 e N sendo N um número inteiro fornecido pelo usuário. O programa deverá mostrar também o número de divisões que ele executou para encontrar os números primos. Serão avaliados o funcionamento, o estilo e o número de testes (divisões) executados.**

**Faça um programa que calcule o mostre a média aritmética de N notas.**

In [ ]:
n = int(input('Informe a quantidade de notas: '))
soma = 0

for i in range(n):
  nota = int(input('Informe a nota: '))
  soma += nota

media = soma / n

print(f'A média das notas é: {media}')

Informe a quantidade de notas: 3
Informe a nota: 5
Informe a nota: 6
Informe a nota: 7
A média das notas é: 6.0


**Faça um programa que peça para n pessoas a sua idade, ao final o programa devera verificar se a média de idade da turma varia entre 0 e 25,26 e 60 e maior que 60; e então, dizer se a turma é jovem, adulta ou idosa, conforme a média calculada.**

In [ ]:
n = int(input('Informe a quantidade de pessoas: '))

for i in range(n):
  idade = int(input('Informe a idade: '))
  soma += idade

media = soma / n

if media >= 0 and media <= 25:
  print('Turma jovem')
elif media >= 26 and media <= 60:
  print('Turma adulta')
else:
  print('Turma idosa')

Informe a quantidade de pessoas: 3
Informe a idade: 12
Informe a idade: 13
Informe a idade: 18
Turma jovem


**Numa eleição existem três candidatos. Faça um programa que peça o número total de eleitores. Peça para cada eleitor votar e ao final mostrar o número de votos de cada candidato.**

In [ ]:
n_eleitores = int(input('Informe o numero total de eleitores: '))
c1 = 0
c2 = 0
c3 = 0

for i in range(n_eleitores):
  voto = int(input('Informe o seu voto: (1, 2 ou 3) '))
  if voto == 1:
    c1 += 1
  elif voto == 2:
    c2 += 1
  elif voto == 3:
    c3 += 1
  else:
    print('Voto inválido')

print(f'O candidato 1 teve {c1} votos\nO candidato 2 teve {c2} votos\nO candidato 3 teve {c3} votos')

Informe o numero total de eleitores: 10
Informe o seu voto: (1, 2 ou 3) 1
Informe o seu voto: (1, 2 ou 3) 2
Informe o seu voto: (1, 2 ou 3) 3
Informe o seu voto: (1, 2 ou 3) 1
Informe o seu voto: (1, 2 ou 3) 2
Informe o seu voto: (1, 2 ou 3) 3
Informe o seu voto: (1, 2 ou 3) 2
Informe o seu voto: (1, 2 ou 3) 1
Informe o seu voto: (1, 2 ou 3) 1
Informe o seu voto: (1, 2 ou 3) 2
O candidato 1 teve 4 votos
O candidato 2 teve 4 votos
O candidato 3 teve 2 votos


**Faça um programa que calcule o número médio de alunos por turma. Para isto, peça a quantidade de turmas e a quantidade de alunos para cada turma. As turmas não podem ter mais de 40 alunos.**

In [ ]:
n_turmas = int(input('Informe a quantidade de turmas: '))
soma = 0

for i in range(n_turmas):
  n_alunos = int(input('Informe a quantidade de alunos: '))
  if n_alunos >= 40:
    print('Quantidade inválida')
  else:
    soma += n_alunos

media = soma / n_turmas
print(f'Média de {media} alunos por turma')

Informe a quantidade de turmas: 5
Informe a quantidade de alunos: 20
Informe a quantidade de alunos: 20
Informe a quantidade de alunos: 20
Informe a quantidade de alunos: 20
Informe a quantidade de alunos: 20
Média de 20.0 alunos por turma


**Faça um programa que calcule o valor total investido por um colecionador em sua coleção de CDs e o valor médio gasto em cada um deles. O usuário deverá informar a quantidade de CDs e o valor para em cada um.**

In [ ]:
n = int(input('Informe a quantidade de CDs: '))
soma = 0

for i in range(n):
  cd = float(input('Informe o valor do CD: '))
  soma += cd

media = soma / n

print(f'O total investido foi: {soma}')
print(f'A média gasta em cada CD foi: {media}')

Informe a quantidade de CDs: 5
Informe o valor do CD: 12.5
Informe o valor do CD: 15
Informe o valor do CD: 14.99
Informe o valor do CD: 13.76
Informe o valor do CD: 11
O total investido foi: 67.25
A média gasta em cada CD foi: 13.45


**O Sr. Manoel Joaquim possui uma grande loja de artigos de R$ 1,99, com cerca de 10 caixas. Para agilizar o cálculo de quanto cada cliente deve pagar ele desenvolveu um tabela que contém o número de itens que o cliente comprou e ao lado o valor da conta. Desta forma a atendente do caixa precisa apenas contar quantos itens o cliente está levando e olhar na tabela de preços. Você foi contratado para desenvolver o programa que monta esta tabela de preços, que conterá os preços de 1 até 50 produtos, conforme o exemplo abaixo:**

```
Lojas Quase Dois - Tabela de preços
1 - R$ 1.99
2 - R$ 3.98
...
50 - R$ 99.50
```



In [ ]:
preco = 1.99
for i in range(1, 51):
  print(f'{i} - R$ {preco:.2f}')
  preco += 1.99

1 - R$ 1.99
2 - R$ 3.98
3 - R$ 5.97
4 - R$ 7.96
5 - R$ 9.95
6 - R$ 11.94
7 - R$ 13.93
8 - R$ 15.92
9 - R$ 17.91
10 - R$ 19.90
11 - R$ 21.89
12 - R$ 23.88
13 - R$ 25.87
14 - R$ 27.86
15 - R$ 29.85
16 - R$ 31.84
17 - R$ 33.83
18 - R$ 35.82
19 - R$ 37.81
20 - R$ 39.80
21 - R$ 41.79
22 - R$ 43.78
23 - R$ 45.77
24 - R$ 47.76
25 - R$ 49.75
26 - R$ 51.74
27 - R$ 53.73
28 - R$ 55.72
29 - R$ 57.71
30 - R$ 59.70
31 - R$ 61.69
32 - R$ 63.68
33 - R$ 65.67
34 - R$ 67.66
35 - R$ 69.65
36 - R$ 71.64
37 - R$ 73.63
38 - R$ 75.62
39 - R$ 77.61
40 - R$ 79.60
41 - R$ 81.59
42 - R$ 83.58
43 - R$ 85.57
44 - R$ 87.56
45 - R$ 89.55
46 - R$ 91.54
47 - R$ 93.53
48 - R$ 95.52
49 - R$ 97.51
50 - R$ 99.50
